perf-testing/test_cloudwatch.rb

```diff
@@ -94,32 +85,36 @@ def separate_byte_data(data)
   separated
 end
 
-def write_test_output(operation, protocol, dimension, metric, input, outfile)
+def format_test_output(operation, protocol, dimension, metric, input, output, iterations)
   input.sort!
   result = {
-    "service": 'CloudWatch',
-    "test_case": operation,
-    "protocol": protocol,
-    "dimension_value": dimension,
-    "metric": metric,
-    "p50": p50(input),
-    "p90": p90(input),
-    "max": input.last,
-    "n": ITERATIONS
+    service: 'CloudWatch',
+    test_case: operation,
+    protocol: protocol,
+    dimension_value: dimension,
+    metric: metric,
+    p50: p50(input),
+    p90: p90(input),
+    max: input.last,
+    n: iterations
   }
-  outfile.puts(JSON.pretty_generate(result))
+  output << result
```

Comment:
Nice, I like this over writing each time to the output file.

Generally however I'd lean towards not modifying the arguments: return the result and append it outside the call.